### PR TITLE
Upgrade geos and expat

### PR DESCRIPTION
**Pull request by shrakesh**
Upgrade geos and expat

Following are the upgrade:
Upgraded geos version to 3.10.2
Upgraded expat version to 2.4.4
Upgraded script to support listed changes.

Following are the commits which is patched from postgis repo
to support geos 3.10.2.

https://github.com/postgis/postgis/commit/9e90d70b6ad289fa7e298f1173ff1254d558a004
https://trac.osgeo.org/postgis/changeset/ec7f996bc9537919ea900f2517ab6ceae78c2b70/git
https://github.com/postgis/postgis/commit/9d3a207b5777860813cd935ab829292bceb2fcb8
https://github.com/postgis/postgis/commit/ad333941d847141219f753e49a2ff468339a4acc

also uncommented code which is supported geos 3.5 and later version.

### CODE REVIEW

**Comment by shrakesh** [comment]
Pipeline: https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/postgis-master-gpdb6_cont
rhel 7: s3://postgis-gppkg/postgis-2.5.4+pivotal.6-gp6-rhel7-x86_64.gppkg
rhel8: s3://postgis-gppkg/postgis-2.5.4+pivotal.6-gp6-rhel8-x86_64.gppkg

File: postgis/build/postgis-2.5.4/liblwgeom/lwgeom_geos.c [hunk 1]
```diff
@@ -870,41 +917,41 @@ lwgeom_union(const LWGEOM* geom1, const LWGEOM* geom2)
...
+ 	result->srid = geom1->srid;
 
-// 	return result;
-// }
+ 	return result;
+ }
 
 /* ------------ BuildArea stuff ---------------------------------------------------------------------{ */
-
+#if POSTGIS_GEOS_VERSION < 38
```
[the remark this answers is not in the document]
Comment (shrakesh) [reply postgis/build/postgis-2.5.4/liblwgeom/lwgeom_geos.c:954]: @orhankislal  yes this from ustream, we are not exactly disabled the code but created 2 branch 1 for older geos and other for new geos. there were ~3 upstream marges  i can create 3 different commits for that and 1 for uncomment our old changes. would that be a good idea?

[the remark this answers is not in the document]
Comment (orhankislal) [reply postgis/build/postgis-2.5.4/liblwgeom/lwgeom_geos.c:954]: I think that'll work. My concern was mostly about the changed test results. If they come from upstream, I think we are good to go. If we change them for greenplum, we have to be careful to justify them.

[the remark this answers is not in the document]
Comment (shrakesh) [reply postgis/build/postgis-2.5.4/liblwgeom/lwgeom_geos.c:954]: > I think that'll work. My concern was mostly about the changed test results. If they come from upstream, I think we are good to go. If we change them for greenplum, we have to be careful to justify them.

All of the changes(test and new features basically in postgis) in this pr is coming from upstream. haven't made any changes specific to greenplum.  I will breakdown this pr into logical commits.

**Review by orhankislal — APPROVED** [review]

**Pull request merged**